### PR TITLE
Update `azurerm_kubernetes_cluster` to allow the linux_profile ssh key to be rotated

### PR DESCRIPTION
**Pull request by implodingduck**
Changing SSH key for aks nodes no longer requires a forceNew:
https://learn.microsoft.com/en-us/azure/aks/node-access#update-ssh-key-on-an-existing-aks-cluster-preview

so this pull request removes the force new flag and adds a testcase to change the ssh key

### CODE REVIEW

File: internal/services/containers/kubernetes_cluster_other_resource_test.go [hunk 1]
```diff
@@ -140,6 +140,43 @@ func TestAccKubernetesCluster_linuxProfile(t *testing.T) {
...
+			Config: r.linuxProfileConfig(data, "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCqaZoyiz1qbdOQ8xEf6uEu1cCwYowo5FHtsBhqLoDnnp7KUTEBN+L2NxRIfQ781rxV6Iq5jSav6b2Q8z5KiseOlvKA/RF2wqU0UPYqQviQhLmW6THTpmrv/YkUCuzxDpsH7DUDhZcwySLKVVe0Qm3+5N2Ta6UYH3lsDf9R9wTP2K/+vAnflKebuypNlmocIvakFWoZda18FOmsOoIVXQ8HWFNCuw9ZCunMSN62QGamCe3dL5cXlkgHYv7ekJE15IA9aOJcM7e90oeTqo+7HTcWfdu0qQqPWY5ujyMw/llas8tsXY85LFqRnr3gJ02bAscjc477+X+j/gkpFoN1QEmt terraform@demo.tld"),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("kube_config.0.client_key").Exists(),
+				check.That(data.ResourceName).Key("kube_config.0.client_certificate").Exists(),
+				check.That(data.ResourceName).Key("kube_config.0.cluster_ca_certificate").Exists(),
+				check.That(data.ResourceName).Key("kube_config.0.host").Exists(),
+				check.That(data.ResourceName).Key("kube_config.0.username").Exists(),
+				check.That(data.ResourceName).Key("kube_config.0.password").Exists(),
+				check.That(data.ResourceName).Key("linux_profile.0.admin_username").Exists(),
```
Comment (stephybun) [inline internal/services/containers/kubernetes_cluster_other_resource_test.go:158]: We can omit these explicit checks for this test
```suggestion
```

File: internal/services/containers/kubernetes_cluster_other_resource_test.go [hunk 2]
```diff
@@ -140,6 +140,43 @@ func TestAccKubernetesCluster_linuxProfile(t *testing.T) {
...
+			Config: r.linuxProfileConfig(data, "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDuiot/laqt5Bxhy0Kaj+g9GqBkr+PEjB18ty59MZC+QXmG7pgOb98FT3BrHUoCtqWWqUASMyMjKIL1fR9HCsV5hJdiecuksldGoWIg9Idr6+5hYDBpPJrm/JHbQBf259YfEi8pQtzAL1ppAv/FqL//MZo2vztnrPi5yDWl+G7ItzF5XeURPtG9DC97T9stUOeLl8bqM4X2ZtydsCkoCia/tajBcp1dFf13kZL9SCsxW452fgXGipx/LvSQQAFcT+xt6mZSgFfSCHUZm4JnxLjSweefRpIOxkO09QS4BlK1MUeBx/EL1Cxv4ql0Uu5x9m64G8E3m1PIJabLp/8Dw84d"),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("kube_config.0.client_key").Exists(),
+				check.That(data.ResourceName).Key("kube_config.0.client_certificate").Exists(),
+				check.That(data.ResourceName).Key("kube_config.0.cluster_ca_certificate").Exists(),
+				check.That(data.ResourceName).Key("kube_config.0.host").Exists(),
+				check.That(data.ResourceName).Key("kube_config.0.username").Exists(),
+				check.That(data.ResourceName).Key("kube_config.0.password").Exists(),
+				check.That(data.ResourceName).Key("linux_profile.0.admin_username").Exists(),
```
Comment (stephybun) [inline internal/services/containers/kubernetes_cluster_other_resource_test.go:172]: Same here 
```suggestion
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this PR @implodingduck. Two minor comments, also the docs need to be updated. Once that's done this should be good to go 👍

**Comment by implodingduck** [comment]
Thanks @stephybun, I have made the changes that you requested.

**Review by stephybun — APPROVED** [review]
Thanks @implodingduck LGTM 🌮

**Pull request merged**